src/components/application_manager/src/message_helper.cc

```diff
@@ -2395,24 +2395,21 @@ mobile_apis::Result::eType MessageHelper::ProcessSoftButtons(
     }
 
     switch (request_soft_buttons[i][strings::type].asInt()) {
-      case SoftButtonType::SBT_IMAGE: {
-        // Any text value for type "IMAGE" should be ignored.
-        if (request_soft_buttons[i].keyExists(strings::text)) {
-          request_soft_buttons[i].erase(strings::text);
-        }
-
-        if ((!request_soft_buttons[i].keyExists(strings::image) ||
-             (Result::SUCCESS !=
-              VerifyImage(
-                  request_soft_buttons[i][strings::image], app, app_mngr)))) {
-          return Result::INVALID_DATA;
-        }
-        break;
-      }
+      case SoftButtonType::SBT_IMAGE:
       case SoftButtonType::SBT_TEXT: {
-        if ((!request_soft_buttons[i].keyExists(strings::text)) ||
-            (!VerifySoftButtonString(
-                 request_soft_buttons[i][strings::text].asString()))) {
+        Result::eType verification_result = Result::INVALID_DATA;
+        verification_result =
+            VerifyImage(request_soft_buttons[i][strings::image], app, app_mngr);
```

Comment:
@indlu, this construction looks like overhead. Better to write like this 
`const Result::eType verification_result = VerifyImage(request_soft_buttons[i][strings::image], app, app_mngr);`